staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -159,7 +159,9 @@ func AggregateContainerRequests(pod *v1.Pod, opts PodResourcesOptions) v1.Resour
 		containerReqs := container.Resources.Requests
 		if opts.UseStatusResources {
 			cs, found := containerStatuses[container.Name]
-			if found && cs.Resources != nil {
+			if found &&
+				container.Resources.Requests != nil &&
+				cs.Resources != nil && cs.Resources.Requests != nil {
 				containerReqs = determineContainerReqs(pod, &container, cs)
```

Comment:
Shall we add the check inside the `determineContainerReqs` function? Also need to check for `container`/`container.Resources` seems
https://github.com/kubernetes/kubernetes/blob/e44ddbf1dae532b00999970ffe078c2d06920309/staging/src/k8s.io/component-helpers/resource/helpers.go#L227-L232.

Do we also need to update kubectl as it uses the same logic https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/util/resource/resource.go#L57,L71

Reply:
>Do we need to update kubectl as well https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/util/resource/resource.go#L57,L71

Good find, I will send a follow-up PR after this one gets merged to scoped down the required approvers.